report/metrics.go

```diff
@@ -49,13 +51,26 @@ type Sample struct {
 
 // MakeMetric makes a new Metric.
 func MakeMetric() Metric {
-	return Metric{}
+	return Metric{
+		Samples: ps.NewList(),
+	}
+}
+
+// Copy returns a value copy of the Metric. Metric is immutable, so we can skip
+// this.
+func (m Metric) Copy() Metric {
+	return m
 }
 
 // WithFirst returns a fresh copy of m, with First set to t.
 func (m Metric) WithFirst(t time.Time) Metric {
-	m.First = t
-	return m
+	return Metric{
+		Samples: m.Samples,
+		Max:     m.Max,
+		Min:     m.Min,
+		First:   t,
+		Last:    m.Last,
+	}
```

Comment:
Thinking about it, I think that might have been ps's motivation for putting everything behind an interface?